File: src/main/java/com/esprit/pi_project/entities/Club.java
package com.esprit.pi_project.entities;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.Data;

import java.io.Serializable;
import java.util.List;
@Entity
@Data
public class Club implements Serializable {
    @Id
    @GeneratedValue(strategy= GenerationType.IDENTITY)
    private Long clubId;
    private String clubName;
    private String clubCategory;
    //private User manager;
    private String description;
    private Integer membershipCount;
    @Enumerated(EnumType.STRING)
    private Tag tag;
    @OneToMany(cascade = CascadeType.ALL)
    @JsonIgnore
    private List<Post> posts;


}

File: Frontend/src/app/interfaces/club.ts
import { UserModel } from "../models/userModel";

export interface Club{
  clubId?: number;
  
  clubName?: string;
  // manager?: User; // Assuming User interface is defined elsewhere
  description?: string;
  membershipCount?: number;
  tag?: string;
  image?: any;
  user?: UserModel;
}
